src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,134 @@
+        String command = input.split(" ")[0];
+        InputCommand inputCommand;
+
+        try {
+            inputCommand = InputCommand.valueOf(command.toUpperCase());
+        } catch (Exception e) {
+            inputCommand = InputCommand.INVALID;
+        }
+
+        switch (inputCommand) {
+        case BYE:
+            return new ExitCommand();
+        case LIST:
+            return new ListCommand();
+        case DONE:
+            return new DoneCommand(input);
+        case DELETE:
+            return new DeleteCommand(input);
+        case TODO:
+        case DEADLINE:
+        case EVENT:
+            return new AddCommand(input, inputCommand);
+        case GET_TASK:
+            return new GetCommand(input);
+        case FIND:
+            System.out.println("sdfs");
+            return new FindCommand(input);
+        default:
+            throw new DukeException("Give me a valid banana (input)!");
+        }
+
+    }
+
+    /**
+     * Checks if the input has valid index.
+     * @param input String input of user.
+     * @param tasksNumber Number of tasks.
+     * @return true if valid, false if invalid.
+     */
+    public static boolean isValidIndex(String input, int tasksNumber) {
+        String[] stringArray = input.split(" ");
+        int index;
+        try {
+            index = Integer.parseInt(stringArray[1]);
+        } catch(Exception e) {
+            return false;
+        }
+        return index <= tasksNumber && index > 0;
+    }
+
+    /**
+     * Gets index of input.
+     * @param input String input of user.
+     * @return Index in input.
+     */
+    public static int getIndex(String input) {
+        String[] stringArray = input.split(" ");
+        return Integer.parseInt(stringArray[1]) - 1;
+    }
+
+    /**
+     * Gets description of Todo object.
+     * @param input String input of user.
+     * @return String description of Todo object.
+     * @throws DukeException if Todo description is empty.
+     */
+    public static String getTodoDescription(String input) throws DukeException{
+        try {
+            return input.substring(5);
+        } catch (Exception e){
+            throw new DukeException("Todo cannot be empty!");
+        }
+    }
+
+    /**
+     * Gets array of deadline description strings.
+     * @param input String input of user.
+     * @return Array of deadline description strings.
+     */
+    public static String[] getDeadlineStrings(String input) {
+        String[] stringArray = input.split(" /by ");
+        stringArray[0] = stringArray[0].substring(9);
+        return stringArray;
+    }
+
+    /**
+     * Gets array of event description strings.
+     * @param input String input of user.
+     * @return Array of event description strings.
+     */
+    public static String[] getEventTimeStrings(String input) {
+        String[] stringArray = input.split(" /at ");
+        stringArray[0] = stringArray[0].substring(6);
+        return stringArray;
+    }
```

Comment:
I think it's really great that you abstract this to methods! It makes debugging easier (you might want to avoid magic number though!)